java/gateway-core/src/main/java/com/avaulta/gateway/rules/transforms/RecordTransform.java

```diff
@@ -0,0 +1,73 @@
+package com.avaulta.gateway.rules.transforms;
+
+import com.fasterxml.jackson.annotation.*;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+
+
+/**
+ *
+ * a transform to apply within a record
+ *
+ * yaml representation
+ *  - !<redact>
+ *    path: "foo"
+ *
+ * would prefer that both
+ *
+ *  - redact: "foo"
+ *
+ * and
+ *  - redact
+ *    path: "foo"
+ *
+ * would be supported as equivalent ... but can't get that to work with Jackson YAML
```

Comment:
That *could* be done with a custom serializer, but I think as it is now is fine as we see the subtypes and the fields.